### PR TITLE
Updated deprecated dependency request-promise-native

### PR DESCRIPTION
**Pull request by yashkohli88**
The crawler code has been updated to replace the deprecated dependency, **request-promise-native**, with **axios**. The following tasks have been performed:

 - [x] A fetch.js file was created in the lib folder. This file contains a function which takes request parameters as input, uses the axios library to initiate HTTP requests, and returns responses. It accepts method, URL, responseType ('json', 'text', 'stream'), headers, and data as request parameters. 
 - [x] This function was integrated into the files in lib/providers/fetch folder wherever the request-promise-native library was previously invoked. Various files and functions were updated to accommodate this change.
 - [x] All replaced statements were individually tested, and an integration test suite was also used to ensure that no functionality was broken due to these changes.
 - [x] Test files were updated where stubs had been declared using request-promise-native. 
 - [x] The request-promise-native library was removed from the package.json file.
 - [x] A test file for fetch.js, named fetchTests.js, was created to test the function with various scenarios. Please note that an internet connection is necessary to run these test cases.

#### Future scope - 
- Use this same file and function to replace **request-promise-native** dependency in service repo as well
- Replace **request** and **request-retry** library with **axios** and **axios-retry**
- Refactor the code to declare the headers in centralised fetch.js file

This PR is related to the issue https://github.com/clearlydefined/crawler/issues/555

### CODE REVIEW

File: providers/store/webhookDeltaStore.js [hunk 1]
```diff
@@ -19,17 +19,17 @@ class WebhookDeltaStore {
...
       },
       resolveWithFullResponse: true
     }
     try {
-      const response = await request(options)
-      if (response.statusCode !== 200)
-        this.logger.info(`Failure  Firing webhook failed: ${response.statusCode} ${response.statusMessage}`)
+      const response = await callFetch(options)
+      if (response.status !== 200)
+        this.logger.info(`Failure  Firing webhook failed: ${response.status} ${response.statusText}`)
```
Comment (yashkohli88) [inline providers/store/webhookDeltaStore.js:32]: This function could not be tested as I was not able to find the correct flow to hit this function.

File: providers/fetch/podFetch.js [hunk 2]
```diff
@@ -89,8 +89,9 @@ class PodFetch extends AbstractFetch {
   async _getSourceArchive(dir, url, podspec) {
     const archive = path.join(dir.name, `${podspec.name}-${podspec.version}.archive`)
     const output = path.join(dir.name, `${podspec.name}-${podspec.version}`)
+    const response = await callFetch({ url: url, responseType: 'stream' })
     return new Promise((resolve, reject) => {
-      request({ url: url, json: false, encoding: null }).pipe(
+      response.pipe(
```
Comment (yashkohli88) [inline providers/fetch/podFetch.js:94]: This function could not be tested as I was not able to find the pod component without github url detail which would have triggered this function.

Comment (qtomlinson) [reply providers/fetch/podFetch.js:94]: The above code path can be hit using one of [these coordinates](https://github.com/user-attachments/files/15539731/dev-hasHttpSource-coordinates.txt).

File: providers/store/webhookDeltaStore.js (hunk 1 above)
Comment (qtomlinson) [reply providers/store/webhookDeltaStore.js:32]: setup to test the webhookDeltaStore locally: 
- in .env file, set the following env variables:
  - CRAWLER_STORE_PROVIDER=cdDispatch+cd(file)+webhook # test webhook
  - CRAWLER_WEBHOOK_URL="http://service:4000/webhook"
  - CRAWLER_WEBHOOK_TOKEN="Bearer github_token_here"
  - WEBHOOK_CRAWLER_SECRET="Bearer github_token_here"

- trigger a harvest in the crawler and then upon completion of harvest the highlighted code will be executed.  
- To ensure the message has been correctly transmitted, can also set a break point at [handleCrawlerCall](https://github.com/clearlydefined/service/blob/e07cf237ed5d4f6ec12a0cb383caf41b00050a9c/routes/webhook.js#L58) in service.  Verify that the response status is set to 200.

File: providers/fetch/podFetch.js (hunk 2 above)
Comment (yashkohli88) [reply providers/fetch/podFetch.js:94]: Thank you @qtomlinson for the list of coordinates. It was very helpful and this functionality has been covered in the testing now. We can add one of the version of pod/cocoapods/-/xcbeautify coordinate to our integration test cases as the size of this particular component is least out of all the coordinates that I tested.

File: providers/store/webhookDeltaStore.js (hunk 1 above)
Comment (yashkohli88) [reply providers/store/webhookDeltaStore.js:32]: Thank you @qtomlinson for taking out time to help me with testing this portion of the code change. I followed these steps and I am able to test the remaining portion as well.

File: providers/fetch/cratesioFetch.js [hunk 3]
```diff
@@ -68,13 +68,16 @@ class CratesioFetch extends AbstractFetch {
...
   // Example: https://crates.io/api/v1/crates/bitflags/1.0.4/download
   async _getPackage(zip, version) {
+    const response = await callFetch({
+      url: `https://crates.io${version.dl_path}`,
+      responseType: 'stream',
+      headers: {
+        'User-Agent': 'clearlydefined.io crawler (clearlydefined@outlook.com)',
+        Accept: 'text/html'
+      }
+    })
```
[the remark this answers is not in the document]
Comment (qtomlinson) [reply providers/fetch/cratesioFetch.js:78]: @ajhenry Your feedback is much appreciated!

[the remark this answers is not in the document]
Comment (ajhenry) [reply providers/fetch/cratesioFetch.js:78]: This makes sense to me as we're essentially mimicking what a web crawler would request 👍

[the remark this answers is not in the document]
Comment (qtomlinson) [reply providers/fetch/cratesioFetch.js:78]: Thanks for the confirmation!

File: lib/fetch.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      method: request.method,
+      url: request.url,
+      responseType: request.responseType,
+      headers: request.headers,
+      data: request.body
+    })
+    if (request.resolveWithFullResponse) return response
+    return response.data
+  } catch (error) {
+    return error.response
```
Comment (qtomlinson) [inline lib/fetch.js:18]: log and rethrow? catch blocks currently in calling code.

**Review by lumaxis — APPROVED** [review]
Makes sense to me! Should also make it fairly easy to move to `fetch` eventually, if we want to 🚀

**Pull request merged**